packages/generator/ni_measurement_plugin_sdk_generator/client/templates/measurement_plugin_client.py.mako

```diff
@@ -183,7 +190,30 @@ class ${class_name}:
             Stream of measurement output.
         """
         parameter_values = [${measure_api_parameters}]
-        request = self._create_measure_request(parameter_values)
-
-        for response in self._get_stub().Measure(request):
-            yield self._deserialize_response(response)
+        with self._initialization_lock:
+            if self._measure_response is not None:
+                raise RuntimeError("A measure call is already in progress.")
+            request = self._create_measure_request(parameter_values)
+            self._measure_response = self._get_stub().Measure(request)
+
+        try:
+            for response in self._measure_response:
+                % if output_metadata:
+                yield self._deserialize_response(response)
+                % else:
+                yield
+                % endif
+        except grpc.RpcError as e:
+            if e.code() == grpc.StatusCode.CANCELLED:
+                print("The measure call is canceled.")
```

Comment:
Don't print to stdout. This is library code, so it shouldn't write random stuff to stdout. Use the logger if you want to print a debug message.